src/protocolsupport/protocol/packet/middle/clientbound/play/MiddleWorldCustomSound.java

```diff
@@ -23,7 +24,15 @@ public MiddleWorldCustomSound(ConnectionImpl connection) {
 
 	@Override
 	public void readFromServerData(ByteBuf serverdata) {
-		id = StringSerializer.readString(serverdata, ProtocolVersionsHelper.LATEST_PC);
+		if(connection.getVersion() == ProtocolVersionsHelper.LATEST_PE)
```

Comment:
can do *connection.getVersion().isPE()* here, keeps future-proofing. can add conditional for version specific laters if/when they come up

Reply:
also, you'll want to modify the PE extended version of this class (protocolsupport.protocol.packet.middleimpl.clientbound.play.v_pe.WorldCustomSound). The middle packets themselves shouldnt have anything version specific- they're basically the interface layer for the native server version.